internal/pkg/agent/configuration/fleet.go

```diff
@@ -19,6 +19,7 @@ type FleetAgentConfig struct {
 	Client              remote.Config      `config:",inline" yaml:",inline"`
 	Info                *AgentInfo         `config:"agent" yaml:"agent"`
 	Server              *FleetServerConfig `config:"server" yaml:"server,omitempty"`
+	FastCheckin         bool               `config:"fast_checkin" yaml:"fast_checkin,omitempty"`
```

Comment:
I think this needs more structure because this will evolve overtime. It will most likely have more settings and things added that effect checkin.

Could you add a new `FleetCheckin` structure and add:


```suggestion
	Checkin         FleetCheckin               `config:"checkin" yaml:"checkin,omitempty"`
```

Then have:

```golang
type FleetCheckin struct {
    Mode string `config:"mode" yaml:"mode,omitempty"`
}

func (f *FleetCheckin) InitDefaults() {
   f.Mode = "standard"
}

func (f *FleetCheckin) Validate() error {
  if f.Mode != "standard" && f.Mode != "on_state_change" {
    return errors.New("check.mode must be either 'standard' or 'on_state_change'")
  }
  return nil
}
```

Reply:
Nice, thank you.